README.md

```diff
@@ -1,5 +1,273 @@
+```
+
+This will create `file1.h5`, `file2.h5` and `file3.h5` in the directory
+`/path/to/output`.
+
+#### Advanced Options
+
+Convert with compression, custom dataset key, and reshape dimensions:
+
+```bash
+mib2h5 -c -d '/rawdata' -r '10x10' -t 300 input.mib
+```
+
+This will:
+- enable Blosc compression
+- store the frames at the dataset key `/rawdata` in the HDF5 file
+- reshape the data to `(10, 10, det_y, det_x)` if there are 100 frames with
+dimensions of `(det_y, det_x)`.
+
+### C API Examples
+
+#### Basic C Example
+
+```c
+#include <mib2h5.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+int main() {
+    // Basic conversion
+    const char* input_files[] = {"input.mib"};
+    int result = mib_to_h5(input_files,    // Input files
+                           1,              // Number of MIB files
+                           NULL,           // NULL to use current directory
+                           true,           // Include metadata
+                           "/data",        // Default key to save frames
+                           "/metadata",    // Default key to save metadata
+                           false,          // NOT to use compression
+                           NULL,           // No reshaping
+                           true,           // Report conversion progress
+                           900             // Seconds to timeout
+                           );
+
+    // Non-zero exit code when there is an error during conversion
+    if (result != 0) {
+        printf("Error: %s\n", mib_to_h5_last_error());
+        return 1;
+    }
+
+    return 0;
+}
+```
+
+#### Advanced C Example
+
+```c
+#include <mib2h5.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+int main() {
+    const char* multiple_files[] = {"file1.mib", "file2.mib", "file3.mib"};
+    int result = mib_to_h5(multiple_files,       // Input files
+                           3,                    // Number of MIB files
+                           "/path/to/output",    // Output directory
+                           true,                 // Include metadata
+                           "/rawdata",           // Custom dataset key
+                           "/meta",              // Custom Metadata key
+                           true,                 // Use compression
+                           "10x10",              // Reshape dimension
+                           true,                 // Report progress
+                           300                   // Seconds to timeout
+                           );
+
+    // Non-zero exit code when there is an error during conversion
+    if (result != 0) {
+        printf("Error: %s\n", mib_to_h5_last_error());
+        return 1;
+    }
+
+    return 0;
+}
+```
+
+### Python API Examples
+
+#### Basic Python Example
+
+```python
+from mib2h5 import convert
+
+# Basic conversion
+try:
+    convert("input.mib")
+except (ValueError, RuntimeError):
+    print("Conversion failed.")
+    raise
```

Comment:
Using `except` catches the exception and saves the user from seeing some nasty traceback. But then using `raise` to re-raise the caught exception reintroduces the nasty traceback (and may make it harder to see the output of the `print` that was added in, if the traceback is long).

Just wanted to raise the question of if it's preferable to save the user from seeing a nasty traceback error or not. If so, doing something alternative to the `raise` may be preferred.

Reply:
Thanks for the suggestion! The idea is to show `convert` will normally raise `ValueError` or `RuntimeError` if there is something wrong during the conversion, and it is up to users how they handle it. Yes you are right the `raise` is kind of redundant here.

Reply:
Haha, I thought about my comment a bit more afterwards: I think it may have been slightly misguided...

In the python example,`mib2h5` is being used as a library, and from the perspective of a user of the library writing their own code, they possibly _would_ want to see a traceback to see where exactly in _their_ code the use of `mib2h5` caused an error, to then have a better chance of debugging what happened when using `mib2h5`.

I was somehow thinking that the python example was code internal to `mib2h5` (which is incorrect!). If that were the case, my comment maybe applies (not wanting internals of `mib2h5` to produce nasty tracebacks upon erroring).

So, I think I got a bit mixed up! But either way, I don't think it affects things drastically :slightly_smiling_face: